### PR TITLE
[1769] Add a user capability in support UI

### PR DESCRIPTION
**Pull request by JonF1**
### Context
* https://trello.com/c/WV1nHbyV/1769-m-add-user
### Changes proposed in this pull request
A form that allows support team to create a user in the UI 
User state set to 'new' before save as not done by default. I think it used to be but state machine was removed from api as per https://github.com/DFE-Digital/teacher-training-api/pull/1427
Feature spec
Link to provider out of scope

### Guidance to review
Comment out `before_action :authenticate` in application controller
visit `/support/users`
click add a user button to be taken to form
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by JonF1** [comment]
[[M] Add user](https://trello.com/c/WV1nHbyV/1769-m-add-user)

File: app/models/user.rb [hunk 1]
```diff
@@ -61,11 +61,17 @@ def notifications_configured?
     user_notifications.count.positive?
   end
 
+  before_create do
```
Comment (JonF1) [inline app/models/user.rb:64]: Is there a better way of doing this? It's just needed for the user controller create action, but a `before_action` wouldn't have worked

File: app/models/user.rb [hunk 2]
```diff
@@ -65,7 +65,7 @@ def notifications_configured?
 
   def email_is_lowercase
     if email.present? && email.downcase != email
-      errors.add(:email, "must be lowercase")
+      errors.add(:email, "The user's email address must be lowercase")
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/models/user.rb:68]: Our errors don't follow the spec 100%, we've made a start by returning errors in a similar structure, eg:

```
{
  "errors": [
    {
      "status": 404,
      "title": "NOT_FOUND",
      "detail": "The requested resource could not be found."
    }
  ]
}
```

but agreed this should probably be left as is so it makes sense when it's returned with the attribute prefixed. @JonF1 any reason why this is being changed?

[the remark this answers is not in the document]
Comment (JonF1) [reply app/models/user.rb:68]: Only to make it more readable. The prefixed attribute isn't being rendered in any of the error messages so the error reads 'must be lowercase'. 
Screenshot 2021-06-01 at 11 11 24

[the remark this answers is not in the document]
Comment (JonF1) [reply app/models/user.rb]: Reverted in 9e83eb

**Review by archferns — APPROVED** [review]

File: app/views/support/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    <%= f.govuk_text_field :first_name, label: { text: "First name", size: "s" }, width: 20 %>
+    <%= f.govuk_text_field :last_name, label: { text: "Last name", size: "s" }, width: 20 %>
+    <%= f.govuk_text_field :email, label: { text: "Email address", size: "s" }, width: 20 %>
+
+    <%= f.govuk_submit %>
+    <% end %>
+
+    <%= govuk_link_to(t("cancel"), support_users_path) %>
+  </div>
+</div>
```
Comment (defong) [inline app/views/support/users/new.html.erb:16]: line ending

**Review by defong — APPROVED** [review]

**Review by dankmitchell — APPROVED** [review]

**Review by dankmitchell — APPROVED** [review]

**Pull request merged**